subprojects/core/src/main/java/org/gradle/api/internal/tasks/AbstractTaskDependency.java

```diff
@@ -16,18 +16,29 @@
 
 package org.gradle.api.internal.tasks;
 
+import org.gradle.api.Action;
 import org.gradle.api.NonNullApi;
 import org.gradle.api.Task;
+import org.gradle.api.internal.artifacts.transform.ArtifactTransformDependency;
 
 import javax.annotation.Nullable;
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.Set;
 
 @NonNullApi
 public abstract class AbstractTaskDependency implements TaskDependencyInternal {
+    private static WorkDependencyResolver<Task> ARTIFACT_TRANSFORM_RESOLVER = new WorkDependencyResolver<Task>() {
```

Comment:
How about we call this `IGNORE_ARTIFACT_TRANSFORM_RESOLVER`?

Reply:
Good point.